test/integration/scheduler/framework_test.go

```diff
@@ -488,6 +526,165 @@ func TestUnreservePlugin(t *testing.T) {
 	}
 }
 
+// TestBindPlugin tests invocation of bind plugins.
+func TestBindPlugin(t *testing.T) {
+	testContext := initTestMaster(t, "bind-plugin", nil)
+	bindPlugin1, bindPlugin2 := &BindPlugin{PluginName: "bind-plugin-1", client: testContext.clientSet}, &BindPlugin{PluginName: "bind-plugin-2", client: testContext.clientSet}
+	// Create a plugin registry for testing. Register an unreserve, a bind plugin and a postBind plugin.
+	registry := framework.Registry{
+		unreservePluginName: NewUnreservePlugin,
+		bindPlugin1.Name(): func(_ *runtime.Unknown, _ framework.FrameworkHandle) (framework.Plugin, error) {
+			return bindPlugin1, nil
+		},
+		bindPlugin2.Name(): func(_ *runtime.Unknown, _ framework.FrameworkHandle) (framework.Plugin, error) {
+			return bindPlugin2, nil
+		},
+		postbindPluginName: NewPostbindPlugin,
+	}
+
+	// Setup initial unreserve and bind plugins for testing.
+	plugins := &schedulerconfig.Plugins{
+		Unreserve: &schedulerconfig.PluginSet{
+			Enabled: []schedulerconfig.Plugin{{Name: unreservePluginName}},
+		},
+		Bind: &schedulerconfig.PluginSet{
+			Enabled: []schedulerconfig.Plugin{{Name: bindPlugin1.Name()}, {Name: bindPlugin2.Name()}},
+		},
+		PostBind: &schedulerconfig.PluginSet{
+			Enabled: []schedulerconfig.Plugin{{Name: postbindPluginName}},
+		},
+	}
+	// Set reserve and bind config for testing
+	pluginConfig := []schedulerconfig.PluginConfig{
```

Comment:
Not related to this PR, but we should make providing a pluginConfig for plugins optional so that something like this is not necessary if one does not need to pass any args to plugins.

Reply:
Agreed.

Reply:
Actually plugin config can be nil. I opened https://github.com/kubernetes/kubernetes/pull/80567 to clean up the verbose code.